challenge_take_out_the_garbage/src/challenge_take_out_the_garbage/take_out_the_garbage.py

```diff
@@ -0,0 +1,94 @@
+# ROS
+import smach
+
+# Robot smach states
+import robot_smach_states as states
+from robocup_knowledge import load_knowledge
+import robot_smach_states.util.designators as ds
+from challenge_take_out_the_garbage.pick_up import PickUpTrash
+from challenge_take_out_the_garbage.drop_down import DropDownTrash
+CHALLENGE_KNOWLEDGE = load_knowledge('challenge_take_out_the_garbage')
+
+
+class TakeOutGarbage(smach.StateMachine):
+    """
+    Takes the garbage out of two trash bins and take it to the drop zone
+    """
+    def __init__(self, robot):
+        """ Initialization method
+
+        :param robot: robot api object
+        """
+        smach.StateMachine.__init__(self, outcomes=["succeeded", "failed", "aborted"])
+
+        # Create designators
+        trashbin_designator = ds.EdEntityDesignator(robot=robot,
+                                                    id=CHALLENGE_KNOWLEDGE.trashbin_id)
+        trash_designator = ds.EntityByIdDesignator(robot=robot, id="trash")
+
+        # Look if there is a second trash bin present
+        if CHALLENGE_KNOWLEDGE.trashbin_id2 is not None:
```

Comment:
If trasbin_id2 is not defined in the knowledge, this crashes, so this check is not working. You could use `hassattr` for that.

Reply:
Do you want to continue in that case? Otherwise, a crash with a clear stack trace would not be too bad, in particular since it's at construction time. This case differs from the previous one, where a dynamic variable is accessed during execution.

Reply:
I think it is designed to do either one or two trashbins, so I think it should continue. Am I right @KevinDang74? In case it should continue and this issue is fixed, Also check https://github.com/tue-robotics/tue_robocup/pull/826/files/0478919eb956778b3eff16854093d1f137d7f7e6#diff-c5260bf72bd92df2256c6b70142eebc0R61. It will crash there again. Because `trashbin_designator2` will not be defined.

Reply:
Fixed it by using the hassattr to check for a second trash bin. If it does not have one it still continues now, because there is only one trash bin in robotics testlabs.